soc/infineon/cat1b/cyw20829/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2023 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_sources(soc.c)
+zephyr_include_directories(.)
+
+# Expedition family defines
+zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1 CY_USING_HAL)
+zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1B COMPONENT_CAT1B)
+zephyr_compile_definitions(COMPONENT_CM33)
+
+# Use custom linker script
+set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/soc/infineon/cat1b/cyw20829/linker.ld CACHE INTERNAL "")
+
+# Generate Application header
+find_program(SREC_CAT srec_cat)
+if(NOT SREC_CAT)
+  message(WARNING "srec_cat not found!")
```

Comment:
opoenocd will return the error during program not patched hex (if skipped post build..)
we use `WARNING` to unblock CI/build run, in documentation we added section about  srec_cat usage.

Reply:
This warning is easily missed and a user risk ending up with a bad hex file without realizing it.

Since the binaries are contiguous, why not use something like `cmake -E cat` followed by `objcopy` instead and remove the need for `srec_cat`?

Reply:
> opoenocd will return the error during program not patched hex (if skipped post build..)

but will the error at this time make it clear to the user that it is due to a missing `srec_cat` ?

> we use WARNING to unblock CI/build run

sound like the docker image should have `srec_cat` should be included in the docker image ?

Reply:
I think adding srec_cat to the docker image and updating this to be an ERROR so that its more clear to the user makes sense.. 

Actually, looking at the https://github.com/zephyrproject-rtos/docker-image/blob/main/Dockerfile.base srecord is already in the list. Can you please confirm that this is the correct repo?